### DISCUSSION TITLE
Warning if use Filtergroup 1

### DISCUSSION THREAD

**Discussion by SKeiler** [StefanMaron/BusinessCentral.LinterCop#687] [New Rule]
We had yesterday an unexpected suprise because a collegue used Filtergroup 1. Filtergroup 1 is an Filter for the whole system. A collegue has set the filter on the item table. 

When you than try to open the item List, there was a lot of items missing (which was filtered out). Also a system restart didn't solve the problem. You had to delete the created cookie.

It is really dangerous to use the Filtergroup 1. So it would be a good idea for a warning.

**Comment by christianbraeunlich** [comment]
Yes, I was just searching for this topic when I found this in the MS Docs: https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/record/record-filtergroup-method#remarks

As is so often the case the message is not highlighted or anything similar making it easier to overlook:
[...] *Dynamics 365 Business Central uses the following filter groups internally. **These filtergroups shouldn't be used**.* [...]

![image](https://github.com/user-attachments/assets/2c9f9c6c-ac4e-4799-92eb-776d007e11d3)

I think that this should not apply to the standard filter group 0.

Any additional thoughts?

**Comment by SKeiler** [comment]
I think the use of the filtergroups -1,0 and 2 are very common. All others will only be used in special cases (and 1 is really dangerous and your hint should be exactly for that filter group)

**Comment by jwikman** [comment]
I have also seen the unintentional use of FilterGroup 1.

I think that a new rule only should warn on FilterGroup 1, no other group. This is because this FilterGroup has a wider scope (all record instances in the current session) than the current record and probably wider than most expect.

Hmm, looking through the docs again... 🤔
I have not yet seen the use of FilterGroup 6 - but I would expect that to have the same scope as FilterGroup 1. Has anyone experience on that one? If that is the case, FilterGroup 6 should also trigger a warning...
